server/aesdsocket.c

```diff
@@ -167,188 +289,253 @@ int receive_and_store_socket_data(int client_fd, int file_fd) {
+        closelog();                // Close syslog
         exit(1);
-    } 
+    }
 
-    if(bind(socket_fd,server_info->ai_addr,server_info->ai_addrlen)==-1)
+    if (bind(socket_fd, server_info->ai_addr, server_info->ai_addrlen) == -1)
     {
-       //Bind operation has failed, log the details,
+        // Bind operation has failed, log the details,
         syslog(LOG_ERR, "Error occured while binding a socket: %s \n", strerror(errno));
         freeaddrinfo(server_info); // free the linked-list
-        closelog(); //Close syslog
-        exit(1); 
+        closelog();                // Close syslog
+        exit(1);
     }
 
-    //Check if daemon to be created
-    if(daemon_mode)
+    // Check if daemon to be created
+    if (daemon_mode)
     {
-        if(!create_daemon())
+        if (!create_daemon())
         {
             syslog(LOG_ERR, "Daemon creation failed, hence exiting");
             freeaddrinfo(server_info); // free the linked-list
-            closelog(); //Close syslog
-            exit(1); 
+            closelog();                // Close syslog
+            exit(1);
         }
     }
 
-     if(listen(socket_fd,20)==-1)
-     {
-        //listen operation has failed, log the details,
+    if (listen(socket_fd, 20) == -1)
+    {
+        // listen operation has failed, log the details,
         syslog(LOG_ERR, "Error occured during listen operation: %s \n", strerror(errno));
         freeaddrinfo(server_info); // free the linked-list
-        closelog(); //Close syslog
-        exit(1); 
-     }
-
+        closelog();                // Close syslog
+        exit(1);
+    }
 
-    file_fd = open(SOCKETDATA_FILE, O_RDWR |O_CREAT | O_TRUNC,0666);
-    if (file_fd == -1) 
+    file_fd = open(SOCKETDATA_FILE, O_RDWR | O_CREAT | O_TRUNC, 0666);
+    if (file_fd == -1)
     {
         syslog(LOG_ERR, "Open/create of /var/tmp/aesdsocketdata failed");
         freeaddrinfo(server_info); // free the linked-list
-        closelog(); //Close syslog
+        closelog();                // Close syslog
         exit(1);
     }
     initialize_sigaction();
     client_addr_size = sizeof(client_addr); // Initialize client address size
-     while(!exit_main_loop)
-     {
-        client_fd = accept(socket_fd, (struct sockaddr *)&client_addr, &client_addr_size);
-        if(client_fd==-1)
+
+    //Create timer thread
+        pthread_t TimerthreadId;
+        ThreadArgs *timerargs = malloc(sizeof(ThreadArgs));
+        if (timerargs == NULL)
         {
-            //accept operation has failed, log the details and continue for next connection
-            syslog(LOG_ERR, "Error occured during accept operation: %s \n", strerror(errno)); 
-            continue;
+            syslog(LOG_ERR, "Failed to allocate memory for thread arguments");
         }
+        else
+        {
+            timerargs->file_fd = file_fd;
+            int err = pthread_create(&TimerthreadId, NULL, timestamp_appender, (void *)timerargs);
```

Comment:
I think you are missing cleanup of TimerthreadId, this is what valgrind is telling you.